docs/contributing/contributing-rules.md

```diff
@@ -1,102 +1,334 @@
+```
+
+##### LOW
+
+Low confidence rules generally find something which appears to be dangerous while reporting a lot of false positives. See examples below:
+
+- https://semgrep.dev/playground/r/php.lang.security.eval-use.eval-use
+- https://semgrep.dev/playground/r/javascript.browser.security.dom-based-xss.dom-based-xss?editorMode=advanced
+
+```
+confidence: LOW 
+```
+
+#### Likelihood
+
+Specify how likely it is that an attacker can exploit the issue that has been found. The possible values are following:
+
+##### HIGH
+
+HIGH likelihood rules tend to be something which is easily reasoned with if found very likely to be a concern:
+
+Some general examples include:
+- The use of weak encryption
+  - https://semgrep.dev/playground/r/go.lang.security.audit.crypto.use_of_weak_rsa_key.use-of-weak-rsa-key?editorMode=advanced
+- Disabled security feature in a configuration
+- Hardcoded secrets that use `"..."`
+  - https://semgrep.dev/playground/r/javascript.jose.security.jwt-hardcode.hardcoded-jwt-secret?editorMode=advanced
+- `taint mode sources` which reach a `taint mode sink` with `taint mode sanitizers`
+
+```
+likelihood: HIGH 
+```
+
+##### MEDIUM
+
+MEDIUM likelihood rules tend to be vulnerable in most circumstances but may be hard for an attacker to achieve due to some constraints. In addition, rules that may only find part of a problem, not the whole issue:
+
+Some general examples:
+- `taint mode sources` which reach a `taint mode sink`  but the source is something which can only be vulnerable in certain conditions e.g. OS Environment Variables, or loading from disk
+  - https://semgrep.dev/playground/r/python.aws-lambda.security.dangerous-spawn-process.dangerous-spawn-process?editorMode=advanced
+- `taint mode sources` with a `taint mode sink` but is missing a `taint mode sanitizer` which can introduce more false positives
+  - https://semgrep.dev/playground/r/javascript.express.security.express-puppeteer-injection.express-puppeteer-injection?editorMode=advanced
+
+```
+likelihood: MEDIUM 
+```
+
+##### LOW
+
+LOW likelihood rules tend to be which tends to be security rule which finds 'something' dangerous, but does not do anything to ensure it is  vulnerable, for example:
+
+- `taint mode sources` such as function arguments which may or may not be tainted which reach a `taint mode sink`
+  - https://semgrep.dev/playground/r/typescript.react.security.audit.react-href-var.react-href-var?editorMode=advanced
+- A rule which uses `search mode` to find the use of a dangerous function e.g. trustAsHTML, bypassSecurityTrust(), eval(), or innerHTML
+  - https://semgrep.dev/playground/r/javascript.browser.security.dom-based-xss.dom-based-xss?editorMode=advanced
+
+```
+likelihood: LOW 
+```
+
+#### Impact
+
+Including an impact indicator tells users and Semgrep App about the rule impact, and how much 'damage' would this vulnerability would cause to the application, we use LOW, MEDIUM, and HIGH indicators:
+
+- SQL Injection vulnerabilities
+- 
```

Comment:
@LewisArdern maybe we missed some other list items here.

Reply:
i think they are explained further below.